packages/hash/external-services/docker-compose.yml

```
@@ -32,31 +36,78 @@ services:
     #   - ../../../var/external-services/opensearch/data:/usr/share/opensearch/data
 
   postgres:
-    image: postgres:14-alpine
+    image: postgres:14-alpine3.16
     deploy:
       restart_policy:
         condition: on-failure
-    env_file:
-      - ../../../.env
-      - ../../../.env.local
     environment:
-      - PGDATA=/data/pgdata
+      PGDATA: /data/pgdata
+      POSTGRES_USER: "${POSTGRES_USER}"
+      POSTGRES_PASSWORD: "${POSTGRES_PASSWORD}"
+      HASH_KRATOS_PG_USER: "${HASH_KRATOS_PG_USER}"
+      HASH_KRATOS_PG_PASSWORD: "${HASH_KRATOS_PG_PASSWORD}"
+      HASH_KRATOS_PG_DATABASE: "${HASH_KRATOS_PG_DATABASE}"
+      HASH_GRAPH_PG_USER: "${HASH_GRAPH_PG_USER}"
+      HASH_GRAPH_PG_PASSWORD: "${HASH_GRAPH_PG_PASSWORD}"
+      HASH_GRAPH_PG_DATABASE: "${HASH_GRAPH_PG_DATABASE}"
     ports:
       - "${POSTGRES_PORT}:5432"
     volumes:
-      - ../../../var/external-services/postgres/data:/data
+      - hash-postgres-data:/var/lib/postgresql/data
       - ./postgres/postgresql.conf:/etc/postgresql/postgresql.conf:ro
-      - ./postgres/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh
+      - ./postgres/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh:ro
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready --username ${POSTGRES_USER}"]
+      interval: 2s
+      timeout: 2s
+      retries: 5
+    command: -c 'config_file=/etc/postgresql/postgresql.conf'
+
+  migration:
```

Could we rename this so that it's easier to distinguish between the containers running the graph migrations and the kratos migrations? If we want to be consistent with the kratos container name:
```suggestion
  graph-migrate:
```

---

Applied in 12b73aedd3da04795e81d4a0a4047578c98cbff4